source/users/sessions.txt

```diff
@@ -0,0 +1,473 @@
+            }
+
+Refresh a User Access Token from a Realm SDK
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Realm SDKs automatically refresh a logged in user's access token if
+the token is expired at the time of a request.
+
+The SDKs also allow you to manually refresh a user's access token by
+calling a method on the ``User`` object.
+
+To learn more, refer to the documentation for your Realm SDK:
+
+- :ref:`Realm C++ SDK <cpp-get-a-user-access-token>`
+- :ref:`Realm Flutter SDK <flutter-access-token>`
+- :ref:`Realm Java SDK <java-get-user-access-token>`
+- :ref:`Realm Kotlin SDK <kotlin-access-token>`
+- :ref:`Realm .NET SDK <dotnet-retrieve-current-user>`
+- :ref:`Realm Node.js SDK <node-get-user-access-token>`
+- :ref:`Realm React Native SDK <react-native-get-user-access-token>`
+- :ref:`Realm Swift SDK <ios-get-a-user-access-token>`
+- :ref:`Web SDK <web-get-user-access-token>`
+
+Configure Refresh Token Expiration
+----------------------------------
+
+By default, refresh tokens expire 60 days after they are issued. You can
+configure this time for your App's refresh tokens to be anywhere between
+30 minutes and 180 days. There are two exceptions:
+
+- Anonymous user refresh tokens do not expire, though anonymous user
+  accounts are deleted after 90 days of inactivity.
+
+- Custom JWT refresh token expiration is determined by the ``exp`` value
+  of the user's JWT.
+
+.. note::
+
+   You can currently only configure refresh token expiration with the
+   Admin API.
+
+Call your App's :admin-api-endpoint:`Set User Refresh Token Expiration
+Time <operation/adminSetRefreshTokenExpiration>` Admin API endpoint with
+the ``expiration_time_seconds`` field of the request body set to your
+desired expiration time.
+
+.. code-block:: bash
+
+   curl -X POST \
+     https://realm.mongodb.com/api/admin/v3.0/groups/{groupId}/apps/{appId}/security/refresh_token_expiration \
+     --data-raw '{
+       "expiration_time_seconds": 864000
+     }'
+
+If you successfully update the refresh token expiration time, the
+endpoint returns a ``204`` response.
+
+Verify & Decode a User Access Token
+-----------------------------------
+
+You can decode an access token to verify that it is well-formed and has
+a valid signature. The decoded response includes other information like
+when the access token expires.
+
+Call your App's :admin-api-endpoint:`Verify & decode an access token
+<operation/adminVerifyUserAccessToken>` Admin API endpoint with the
+token in the ``token`` field of the request body.
```

Comment:
correct me if my understanding of JWTs is correct here. 

while this approach is valid, you could also use any other JWT decoder to do the same thing. for example, i copied the example JWT below into https://jwt.io/ and got the output which you can find [at this link](https://jwt.io/#debugger-io?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJiYWFzX2RldmljZV9pZCI6IjAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMCIsImJhYXNfZG9tYWluX2lkIjoiNWNkYjEyNDA4ZTIzMmFjNGY5NTg3ZmU4IiwiZXhwIjoxNjc2NTExMjgyLCJpYXQiOjE2NzY1MDk0ODIsImlzcyI6IjYzZWQ4MTJhNDNiZTcyYzE3NmFhNWQyMyIsInN0aXRjaF9kZXZJZCI6IjAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMCIsInN0aXRjaF9kb21haW5JZCI6IjVjZGIxMjQwOGUyMzJhYzRmOTU4N2ZlOCIsInN1YiI6IjYzZWQ4MTJhNDNiZTcyYzE3NmFhNWQyMSIsInR5cCI6ImFjY2VzcyJ9.7kHO9wjWvIaD3VewDyPhLyb-oRc7wTYZdD9-hroF-H4).

with that in mind (if my understanding is correct), i think it'd make sense to note that you can use other JWT decoder tools/libraries to achieve the same effect.

Reply:
also can you mention the HTTP status of a successful response here b/c you mention it for unsuccessful (which is 200 🤔)

Reply:
> you could also use any other JWT decoder to do the same thing

Sure you can decode the JWT but you can't validate the signature unless you have the 256 bit secret key. jwt.io tells me that it _might be_ a valid session token but can't confirm that it's authentic.

> can you mention the HTTP status of a successful response

Yeah good catch